### PR TITLE
Adding a database ID to the included sequences

### PR DESCRIPTION
**Pull request by laughedelic**
We need a concept of blast database, which includes its name (it will be passed as arg to `-title` in `makeblastdb`). Given that, we can easily add a reference to the db through

```
>${id}|lcl|${db}
ATTTCGGTACATCATTTCACATTGTAC
```

This way BLAST results will yield something like `00015CA37|lcl|era7bio.db.rna16S` as the value of `sseqid`.

### CODE REVIEW

**Comment by laughedelic** [comment]
@eparejatobes I don't have push permissions in this repo.

**Comment by eparejatobes** [comment]
Done

**Comment by laughedelic** [comment]
@eparejatobes take a look at the code, please.

File: src/main/scala/rna16s.scala [hunk 1]
```diff
@@ -1,49 +1,22 @@
...
-    fasta
-      .parseFastaDropErrors(inputFasta.lines)
-      .filterNot(drop)
-      .map(fa => fa.update(header := FastaHeader(s"${fa.getV(header).id}|${toHeader(db.ncbiID)}")))
-      .appendTo(outputFasta)
-
-    outputFasta
+  val predicate: Row => Boolean = { row =>
+    row(GeneName) == "16s"
+    // TODO: what else?
```
Comment (laughedelic) [inline src/main/scala/rna16s.scala]: what else needs to be checked on filtering?

Comment (eparejatobes) [inline src/main/scala/rna16s.scala]: I just pushed. We also need to filter out short sequences, which requires access to the fasta value.

File: src/main/scala/blastDB.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      println(s"""Downloading the sources...
+        |fasta: ${db.sourceFasta}
+        |table: ${db.sourceTable}
+        |""".stripMargin)
+
+      transferManager.download(db.sourceFasta.bucket, db.sourceFasta.key, sourceFasta.toJava).waitForCompletion
+      transferManager.download(db.sourceTable.bucket, db.sourceTable.key, sourceTable.toJava).waitForCompletion
+    } -&-
+    LazyTry {
+      // FIXME: what is the line separator here?
```
Comment (eparejatobes) [inline src/main/scala/blastDB.scala]: always `'\n'`. See https://github.com/era7bio/rnacentraldb/blob/feature/mirror/rnacentral/bundle/6/src/main/scala/csvUtils.scala#L9-L19

File: src/main/scala/rna16s.scala [hunk 3]
```diff
@@ -1,49 +1,51 @@
...
+    152507  -> "uncultured actinobacterium",
+    1211    -> "uncultured cyanobacterium",
+    153809  -> "uncultured proteobacterium",
+    86027   -> "uncultured beta proteobacterium",
+    86473   -> "uncultured gamma proteobacterium",
+    34034   -> "uncultured delta proteobacterium",
+    56765   -> "uncultured marine bacterium",
+    115414  -> "uncultured marine alpha proteobacterium"
+  )
+
```
Comment (eparejatobes) [inline src/main/scala/rna16s.scala:39]: @rtobes please review that filtering out these taxa is OK.

**Comment by laughedelic** [comment]
This is ready. The generated files are in

```
s3://resources.ohnosequences.com/db/rna16S/
```

the blast-specific files are in the `blastdb/` subfolder.

**Comment by eparejatobes** [comment]
LGTM

File: src/main/scala/rna16s.scala [hunk 4]
```diff
@@ -11,7 +11,7 @@ import rnaCentralTable._
 
 
 case object rna16sDB extends AnyBlastDB {
-  val name = "era7bio.db.rna16s"
+  val name = "era7bio.db.rna16s.fasta"
```
Comment (laughedelic) [inline src/main/scala/rna16s.scala:15]: This will cause problems

**Pull request merged**